### PR TITLE
Added initial set of tests for the cl_khr_semaphore extension

### PR DESCRIPTION
**Pull request by marcat03**

### CODE REVIEW

File: test_conformance/api/test_semaphores.cpp
Comment (bcalidas) [inline test_conformance/api/test_semaphores.cpp:1122]: We have concerns about the use of clFinish() instead of calling clWaitForEvents(signal_1_event); On some implementations clFinish() could deadlock since some of the enqueued commands will not reach complete status until user_event_2 is signaled.

File: test_conformance/api/test_semaphores.cpp
Comment (bcalidas) [inline test_conformance/api/test_semaphores.cpp:1127]: Our concern is that on some implementations signal_2_event status will still be CL_QUEUED and will never be put into the "abnormally terminated" state.

File: test_conformance/api/test_semaphores.cpp
Comment (bcalidas) [inline test_conformance/api/test_semaphores.cpp:953]: recommend testing for "cl_khr_external_semaphore_sync_fd"

**Comment by bcalidas** [comment]
We have some concerns about test_semaphores_invalid_command(). Detailed  Comments inline.

File: test_conformance/api/test_semaphores.cpp
Comment (marcat03) [reply test_conformance/api/test_semaphores.cpp:919]: Agree. I will covert the two test_assert_event_complete into wait for events, leave the delay after and test that signal_1 and wait_1 events are in progress

File: test_conformance/api/test_semaphores.cpp
Comment (marcat03) [reply test_conformance/api/test_semaphores.cpp:953]: Ack

File: test_conformance/api/test_semaphores.cpp
Comment (marcat03) [reply test_conformance/api/test_semaphores.cpp:1122]: ok, I will discuss this issue internally with my team

**Comment by nikhiljnv** [comment]
@kpet @bcalidas 
Should we add back "Focused Review" label here and plan to get this reviewed, approved and merged over next few weeks?

File: test_conformance/api/test_semaphores.cpp
Comment (bcalidas) [inline test_conformance/api/test_semaphores.cpp:88]: change to  "return TEST_SKIPPED_ITSELF".  Using TEST_SKIP here causes the test to return false  failures on implementations that do not support cl_khr_semaphore.

**Comment by bcalidas** [comment]
In its current state this PR is causing the API conformance test to fail for implementations that do not support cl_khr_semaphore.  This is due to the subtests in api/test_semaphores.cpp  using TEST_SKIP as opposed to TEST_SKIPPED_ITSELF after checking for the "cl_khr_semaphore" extension string.

This change needs to be made at several places within the test_semaphores.cpp file. After this edit is made, this PR is good to merge.  We will follow up with another PR that has changes we want.

Thanks,
Balaji Calidas
Qualcomm

File: test_conformance/api/test_semaphores.cpp
Comment (cattanimarco) [reply test_conformance/api/test_semaphores.cpp]: Ack

**Comment by cattanimarco** [comment]
The changes to TEST_SKIP have been made, together with the other trivial fixes

**Review by bcalidas — APPROVED** [review]

**Review by bashbaug — APPROVED** [review]
Merging to unblock additional progress, as discussed in the November 15th teleconference.

**Pull request merged**
